### PR TITLE
catalog: Add static OIDs to builtin objects

### PR DESCRIPTION
**Pull request by jkosh44**
This commit adds static OIDs to builtin objects that are compiled into the binary. In the future, when all OIDs are persisted, this will make dealing with builtin OIDs easier because they will be the same across all environments.

Works towards resolving #25253

### Motivation
This PR adds a known-desirable feature.

### Checklist

- [X] This PR has adequate test coverage / QA involvement has been duly considered.
- [X] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [X] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [X] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [X] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  - Change the OIDs of all builtin objects.

### CODE REVIEW

File: src/adapter/src/catalog/open.rs [hunk 1]
```diff
@@ -553,6 +558,9 @@ impl Catalog {
                         Builtin::Type(_) => unreachable!("loaded separately"),
 
                         Builtin::Func(func) => {
+                            // This OID is never used. `func` has a `Vec` of implementations and
+                            // each implementation has it's own OID. Those are the OIDs that are
+                            // actually used by the system.
```
Comment (benesch) [inline src/adapter/src/catalog/open.rs:563]: Would it be clearer just to pass in 0?

**Review by ParkMyCar — APPROVED** [review]
Nice! What do you think about adding a test to assert the OIDs don't change? Maybe an SLT that checks the output of `SELECT oid FROM mz_objects`?

**Comment by jkosh44** [comment]
> Nice! What do you think about adding a test to assert the OIDs don't change? Maybe an SLT that checks the output of `SELECT oid FROM mz_objects`?

Done

**Comment by ParkMyCar** [comment]
> > Nice! What do you think about adding a test to assert the OIDs don't change? Maybe an SLT that checks the output of `SELECT oid FROM mz_objects`?
> 
> Done

Thank you!

**Pull request merged**